app.py

```diff
@@ -64,19 +62,31 @@ def do_logout():
 
 @app.route('/')
 def home():
+    # get the season leaders' stats that should be stored in database
```

Comment:
There is a lot to unpack here...

1. This is the biggest issue. This is where the engineer part of software engineer comes in, its the hard part -- challenging I should say. Each line here is a trip to the database. Trips to the database are _bad_ in a sense. They take up a lot of time and we always want to minimize them. It may take you some research, and some reworking of your database model, but I'm sure you could get this data in a single query.

2. This code does not belong here (lines 66-75). This code contains the logic on how to interact with the database. This is typically placed in a `repository` or `dal` (short for Data Access Layer) module. Here you would call that repo layer to get your data.
```py
from nba_api.database import repo

@app.route('/')
def home():
    data = repo.get_all_stats()
    if g.user:
        return render_template('search_home.html', data=data)
    # ...
```

3. Very small, possibly just my opinion but I think 4 out of 5 devs would agree (lil joke there)... passing a ton of arguments to a template can bite you. If you get to where your passing a ton of different args to your render function then its good to wrap them in a class/obj and pass only it.
```py
# btw, the only time a class is ever really needed in python
class PlayerStatsData:
    def __init__(self, scoring, rebounding, assisting, stealing, blocking):
        self.scoring = scoring
        self.rebounding = rebounding
        # ...

# ...

data = PlayerStatsData(...)
render_template('page.html', data=data)
```